quinn-proto/src/connection/mod.rs

```diff
@@ -3058,6 +3103,53 @@ impl Connection {
                 self.stats.frame_tx.handshake_done.saturating_add(1);
         }
 
+        // OBSERVED_ADDR
+        let mut send_observed_address =
```

Comment:
Should this be a helper method on `Connection`? Large inline lambdas aren't great for readability.